Dependency to LLVM >= 6.0 in Documentation added

gerl1ng opened the pull request:
**Summary of changes**:

For Kernel Releases >= 4.16 there is a dependency to LLVM >= 6.0 for tools/testing/selftests/bpf
The info to this dependency was added to the documentation / reference guide.

```release-note
<!-- Enter the release note text here if needed -->
```

Commit:
Dependency to LLVM >= 6.0 in Documentation added

For Kernel Releases >= 4.16 there is a dependency to LLVM >= 6.0 for tools/testing/selftests/bpf
The info to this dependency was added to the documentation / reference guide.

borkmann commented on Documentation/bpf.rst:
```diff
@@ -833,7 +833,12 @@ failures:
+.. note:: Since Kernel Release 4.16 the bpf selftests has a dependency on LLVM >= 6.0
+          More info available on some mailing lists. (https://lwn.net/Articles/741773/)
+          Maybe you need to compile LLVM in the current version. More info at
+          http://llvm.org/docs/GettingStarted.html
```
Thanks for the patch! Looks good. Could you also add a clarification why this is the case. The kselftests need this due to BPF to BPF calls. For users not making use of them, this dependency is not needed. Would be great if you could clarify that in the note a bit better so that people don't think LLVM>=6 is required for everything from now on. Thx

Commit:
Dependency to LLVM >= 6.0 in Documentation modified

 - Linked to section with more infos
 - Clarified that not every BPF program has a dependency on LLVM 6.0+

Signed-off by Marius Gerling marius@gerl1ng.de
